### ISSUE TITLE
balanced_accuracy in TabularPredictor.evaluate()

### ISSUE THREAD

**Issue by jwmueller** [awslabs/autogluon#635]
Predictor.evaluate() does not match Predictor.evaluate_predictions() when `eval_metric='balanced_accuracy'`.

Script to reproduce:

```
import autogluon as ag
from autogluon import TabularPrediction as task

train_data = task.Dataset(file_path='https://autogluon.s3.amazonaws.com/datasets/Inc/train.csv')
subsample_size = 500
train_data = train_data.head(subsample_size)
label_column = 'occupation'

test_data = task.Dataset(file_path='https://autogluon.s3.amazonaws.com/datasets/Inc/test.csv')
y_test = test_data[label_column]
test_data_nolabel = test_data.drop(labels=[label_column],axis=1)  # delete label column

metric = 'balanced_accuracy'

output_directory = 'agModels-predictOccupation'  # folder where to store trained models

predictor = task.fit(train_data=train_data, label=label_column, eval_metric=metric,
                     auto_stack=True, output_directory=output_directory,
                     hyperparameters = {'NN':{'num_epochs':5}, 'GBM':{'num_boost_round':100}}, time_limits = 30)

results = predictor.leaderboard(test_data)

y_pred = predictor.predict(test_data_nolabel)
score1 = predictor.evaluate_predictions(y_true=y_test, y_pred=y_pred)

from sklearn.metrics import balanced_accuracy_score
score2 = balanced_accuracy_score(y_test, y_pred)

score3 = predictor.evaluate(test_data)

print(score1, score2, score3)
```

Outputs are:
score1 = 0.1906
score2 = 0.1906 ( = score1)
score 3 = 0.2042
leaderboard weighted_ensemble has score = 0.2042 (= score3)

Here there's new label in the test-data not present in training-data:
```
train_labels = set(train_data[label_column].values) # length = 14
test_labels = set(test_data[label_column].values) # length = 15
print( test_labels.difference(train_labels) )
```
Output: {' Armed-Forces'}

**Comment by jwmueller** [comment]
One bug is in: https://github.com/awslabs/autogluon/blob/master/autogluon/utils/tabular/ml/learner/abstract_learner.py#L237
since -1 is used to replace all new classes in the test data (two different classes will both become -1 values).

Toy example (suppose ['a','b'] are only labels seen in training):
```
from sklearn.metrics import balanced_accuracy_score

y_true = ['a','b','c','d'] # test labels
y_hat = ['a','b','b','a'] # predictions
balanced_accuracy_score(y_true, y_hat) # = 0.5

y_true_cleaned = [0,1,-1,-1] # what labelcleaner would output
y_hat_cleaned = [0,1,1,0] # what labelcleaner would output
balanced_accuracy_score(y_true_cleaned, y_hat_cleaned) # = 0.666
```

**Comment by Innixma** [comment]
This has been fixed.
